app/src/main/java/com/github/libretube/ui/dialogs/AddToPlaylistDialog.kt

```diff
@@ -44,90 +33,51 @@ class AddToPlaylistDialog : DialogFragment() {
+
+            // select the last used playlist
+            lastSelectedPlaylistId?.let { id ->
+                binding.playlistsSpinner.selectedItemPosition = playlists
+                    .indexOfFirst { it.id == id }
+                    .takeIf { it >= 0 } ?: 0
+            }
 
-                    dialog?.hide()
-                    lifecycleScope.launch {
-                        addToPlaylist(playlist.id, playlist.name!!)
-                        dialog?.dismiss()
-                    }
+            msg?.let {
+                with(binding.root.context) {
+                    Toast.makeText(this, getString(it.resId, it.formatArgs), Toast.LENGTH_SHORT).show()
                 }
+                viewModel.onMessageShown()
             }
-    }
 
-    private fun fetchPlaylists(binding: DialogAddToPlaylistBinding) {
-        lifecycleScope.launch {
-            repeatOnLifecycle(Lifecycle.State.CREATED) {
-                playlists = try {
-                    PlaylistsHelper.getPlaylists()
-                } catch (e: Exception) {
-                    Log.e(TAG(), e.toString())
-                    Toast.makeText(context, R.string.unknown_error, Toast.LENGTH_SHORT).show()
-                    return@repeatOnLifecycle
-                }.filter { !it.name.isNullOrEmpty() }
-
-                binding.playlistsSpinner.items = playlists.map { it.name!! }
-
-                if (playlists.isEmpty()) return@repeatOnLifecycle
-
-                // select the last used playlist
-                viewModel.lastSelectedPlaylistId?.let { id ->
-                    binding.playlistsSpinner.selectedItemPosition = playlists
-                        .indexOfFirst { it.id == id }
-                        .takeIf { it >= 0 } ?: 0
-                }
+            saved?.let {
+                dismiss()
+                viewModel.onDismissed()
             }
         }
-    }
-
-    @SuppressLint("StringFormatInvalid")
-    private suspend fun addToPlaylist(playlistId: String, playlistName: String) {
-        val appContext = context?.applicationContext ?: return
-        val streams = videoInfo?.let { listOf(it) } ?: PlayingQueue.getStreams()
 
-        val success = try {
-            if (streams.isEmpty()) throw IllegalArgumentException()
-            PlaylistsHelper.addToPlaylist(playlistId, *streams.toTypedArray())
-        } catch (e: Exception) {
-            Log.e(TAG(), e.toString())
-            appContext.toastFromMainDispatcher(R.string.unknown_error)
-            return
-        }
-        if (success) {
-            appContext.toastFromMainDispatcher(
-                appContext.getString(R.string.added_to_playlist, playlistName)
-            )
-        } else {
-            appContext.toastFromMainDispatcher(R.string.fail)
-        }
+        return MaterialAlertDialogBuilder(requireContext())
+            .setTitle(R.string.addToPlaylist)
+            .setNegativeButton(R.string.createPlaylist) { _, _ ->
```

Comment:
For some reason, pressing this button completely dismisses both dialogs. It's likely that this happens because clicking the dialog button causes the dialog to be dismissed. Thus the death of `AddToPlaylistDialog` causes its `childFragmentManager` to die, which is responsible for displaying `CreatePlaylistDialog`.

That's the reason for the previous workaround with `getButton(DialogInterface.BUTTON_NEGATIVE).setOnClickListener {}`